### PR TITLE
Improve p2

### PR DESCRIPTION
**Pull request by yyuanxuan**

### CODE REVIEW

File: p2/index.qmd [hunk 1]
```diff
@@ -32,27 +33,28 @@ Install the packages by typing in the following commands into RStudio (you do no
 
 If you need to install any of these packages use:
 
-::: {.panel-tabset group="language"}
+::: {.panel-tabset}
```
Comment (Robinlovelace) [inline p2/index.qmd:36]: Why not 'language'?

File: p2/index.qmd [hunk 2]
```diff
@@ -32,27 +33,28 @@ Install the packages by typing in the following commands into RStudio (you do no
...
+```{r} 
 #| eval: false
 if (!require("pak")) install.packages("pak")
 pak::pkg_install(c("sf", "tidyverse", "remotes"))
 # GitHub pkgs
 pak::pkg_install("Nowosad/spDataLarge")
 ```
 
-```{r}
+``` {r}
```
Comment (Robinlovelace) [inline p2/index.qmd:48]: Why the gap here?

File: p2/index.qmd [hunk 3]
```diff
@@ -32,27 +33,28 @@ Install the packages by typing in the following commands into RStudio (you do no
...
 library(sf)          # vector data package 
 library(tidyverse)   # tidyverse packages
 library(spData)  # spatial data package 
 ```
 
 ## Python
 
-```{python, eval=FALSE}
+``` {python}
```
Comment (Robinlovelace) [inline p2/index.qmd:56]: 👍

Comment (Robinlovelace) [reply p2/index.qmd:56]: But without the space I think is better:

```suggestion
```{python}
```

**Review by Robinlovelace — APPROVED** [review]
This looks way better, many thanks @yyuanxuan I just made a couple of minor comments, let's see how it looks on the website.

File: p2/index.qmd (hunk 1 above)
Comment (yyuanxuan) [reply p2/index.qmd:36]: When this option is set, clicking on the Python tab will switch all code blocks to Python. However, due to the different lengths of the R and Python code blocks, this causes the content within the current browser window to suddenly jump, creating significant interference with reading. You have to scroll up or down to find the section you were just reading. Students using Python still need to compare the results of the R code execution, so such page jumps and interruptions can happen frequently. Removing the language grouping can avoid this problem, as each code block switch would only apply to the current tab, resulting in less disruption to the reading experience - in my opinion, this has a better user experience.

File: p2/index.qmd (hunk 3 above)
Comment (yyuanxuan) [reply p2/index.qmd:56]: Thanks for pointing these out. I will fix them in the future— the added space was by mistake and not intentional.

File: p2/index.qmd (hunk 1 above)
Comment (Robinlovelace) [reply p2/index.qmd:36]: OK, great explanation, thanks!

**Pull request merged**